superset-frontend/src/components/Select/utils.ts

```diff
@@ -61,7 +61,7 @@ export function findValue<OptionType extends OptionTypeBase>(
 }
 
 export function getValue(option: string | number | { value: string | number }) {
-  return typeof option === 'object' ? option.value : option;
+  return option && typeof option === 'object' ? option.value : option;
```

Comment:
```suggestion
  return typeof option === 'object' ? option.value : option;
```

Reply:
`typeof option === 'object'` returns true when the value is `null`, that's why the change.
Repl.it: https://replit.com/@DiegoMedina6/BaggyGratefulAdware#index.js

Reply:
The option shouldn't be `null`. The value of an object option is `null`.

Reply:
@diegomedina248 This is happening because the `SelectFilterPlugin` is configured incorrectly. By default, if `options` is a function (async) then `labelInValue` is `true` otherwise is `false`. The `SelectFilterPlugin` is passing a static array of options in the form of `{ label: <label>, value: <value> }` but it's not setting the `labelInValue` to `true`. If you set it to `true`, you will receive the object in the change event. Please make sure to review other parts of the code after changing the `labelInValue` property (like `value`).

Reply:
@diegomedina248 Please disregard my previous message. To support Ant Design's different behavior depending on whether `labelInValue` is active, we need to also account for an option being a primary type. You can keep your last commit where `getValue(option: string | number | { value: string | number | null } | null`